addons/web/static/tests/views/fields/float_field.test.js

```diff
@@ -0,0 +1,483 @@
+        });
+
+        await contains('div[name="float_field"] input').edit("108.2458938598598", {
+            confirm: "blur",
+        });
+        expect(".o_field_widget:eq(0)").toHaveText("108.2458938598598", {
+            message: "The value should not be formatted on blur.",
+        });
+
+        await contains("tr.o_data_row td:not(.o_list_record_selector)").click();
+        await contains('div[name="float_field"] input').edit("18.8958938598598", {
+            confirm: false,
+        });
+        await contains(".o_control_panel_main_buttons .o_list_button_save").click();
+        expect(".o_field_widget:eq(0)").toHaveText("18.8958938598598", {
+            message: "The new value should not be rounded as well.",
+        });
+    }
+);
+
+test("float_field field with placeholder", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        arch: '<form><field name="float_field" placeholder="Placeholder"/></form>',
+    });
+
+    await contains(".o_field_widget[name='float_field'] input").clear();
+    expect(".o_field_widget[name='float_field'] input").toHaveAttribute(
+        "placeholder",
+        "Placeholder"
+    );
+});
+
+test("float field can be updated by another field/widget", async () => {
+    class MyWidget extends Component {
+        static template = xml`<button t-on-click="onClick">do it</button>`;
+        static props = ["*"];
+        onClick() {
+            const val = this.props.record.data.float_field;
+            this.props.record.update({ float_field: val + 1 });
+        }
+    }
+    const myWidget = {
+        component: MyWidget,
+    };
+    registry.category("view_widgets").add("wi", myWidget);
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        arch: `
+            <form>
+                <field name="float_field"/>
+                <field name="float_field"/>
+                <widget name="wi"/>
+            </form>`,
+    });
+
+    await contains(".o_field_widget[name=float_field] input").edit("40");
+
+    expect(".o_field_widget[name=float_field] input:eq(0)").toHaveValue("40.00");
+    expect(".o_field_widget[name=float_field] input:eq(1)").toHaveValue("40.00");
+
+    await contains(".o_widget button").click();
+
+    expect(".o_field_widget[name=float_field] input:eq(0)").toHaveValue("41.00");
+    expect(".o_field_widget[name=float_field] input:eq(1)").toHaveValue("41.00");
+});
+
+test("float field with digits=0", async () => {
+    // This isn't in the orm documentation, so it shouldn't be supported, but
+    // people do it and thus now we need to support it.
+    // Historically, it behaves like "no digits attribute defined", so it
+    // fallbacks on a precision of 2 digits.
+    // We will change that in master s.t. we do not round anymore in that case.
+    Partner._fields.float_field = fields.Float({ string: "Float field", digits: 0 });
```

Comment:
I guess the aim of the test is to verify that it will indeed fall back to the 2 digits precision when digits is set to zero so the test will pass without the line but will also be pointless